scripts/ci/check_compliance.py

```diff
@@ -704,6 +708,38 @@ class KconfigBasicCheck(KconfigCheck):
     def run(self):
         super().run(full=False)
 
+class DevicetreeBindings(ComplianceTest):
+    """
+    Checks if Zephyr devicetree bindings comply with some
+    usual conventions.
+
+    This is NOT an exhaustive check; it should only be
+    considered a 'smoke test'.
+    """
+    name = "DevicetreeBindings"
+    doc = "See https://docs.zephyrproject.org/latest/build/dts/bindings.html for more details."
+    path_hint = ZEPHYR_BASE
+
+    def run(self):
+        bindings_dir = str(Path(ZEPHYR_BASE) / 'dts' / 'bindings')
```

Comment:
> There's already a DevicetreeBindingsCheck calss here that goes over all the binding files and does few checks, can this integrate into that? 

Of course, thanks for catching that :). I botched the rebase on that part. Fixed.

> My main thought is that this filer is only checking the bindings in $ZEPHYR_BASE/dts/bindings, but the buildsystem allows extra bindings on the application directory as well and it would be cool if we could check those too.

Agree, it would be nice, but I don't think the current checks are doing that; are they?

At any rate, if we can get a list of directories to check from the calling environment, we can extend the current code to change `edtlib.bindings_from_dirs([bindings_dir])` to `edtlib.bindings_from_dirs(list_of_bindings_dirs)` very easily.

Can you suggest a way to get the list we need into that method?

Reply:
> > My main thought is that this filer is only checking the bindings in $ZEPHYR_BASE/dts/bindings, but the buildsystem allows extra bindings on the application directory as well and it would be cool if we could check those too.
> 
> Agree, it would be nice, but I don't think the current checks are doing that; are they?

They are right now, check is:

https://github.com/zephyrproject-rtos/zephyr/blob/f09111a2a97aa3457d2a6cf9c362c53133348d3e/scripts/ci/check_compliance.py#L245

so it looks for anything with `dts/bindings` in the path, the buildsystem code is somewhat similar:

https://github.com/zephyrproject-rtos/zephyr/blob/f09111a2a97aa3457d2a6cf9c362c53133348d3e/cmake/modules/dts.cmake#L54-L60

https://github.com/zephyrproject-rtos/zephyr/blob/f09111a2a97aa3457d2a6cf9c362c53133348d3e/cmake/modules/dts.cmake#L141

> At any rate, if we can get a list of directories to check from the calling environment, we can extend the current code to change `edtlib.bindings_from_dirs([bindings_dir])` to `edtlib.bindings_from_dirs(list_of_bindings_dirs)` very easily.
> 
> Can you suggest a way to get the list we need into that method?

I don't think we can cover all cases since one may come up with some weird combination of binding inclusions from multiple roots or custom paths, but I think for the scope of this it may be good enough to load the base bindings + a potential root of any file modified by the patch currently being tested. So we could use the `get_files(filter="d")` list to find all the potential roots (anything under a 'dts/bindings' sub path) and then do the check on "base" and any "base + external root".

Reply:
There's few bindings in the main repository outside of `$ZEPHYR_BASE/dts/bindings`

```
$ find -type f | grep dts/bindings | grep -v '\./dts/bindings/'
./boards/arm/arty/dts/bindings/arm,daplink-qspi-mux.yaml
./boards/arm/efr32bg_sltb010a/dts/bindings/silabs,gecko-wake-up-triggers.yaml
./boards/arm/twr_ke18f/dts/bindings/nxp,flexio.yaml
./boards/arm/actinius_icarus/dts/bindings/actinius-charger-enable.yaml
./boards/arm/actinius_icarus/dts/bindings/actinius-sim-select.yaml
./boards/arm/pinetime_devkit0/dts/bindings/misc/pine64,pinetime-key-out.yaml
./boards/arm/96b_wistrio/dts/bindings/qorvo,rfsw8001.yaml
...
```
And more in `samples` and `tests`.

The current code is considering those as well -- but I'm also interested in having this check usable in an external repository (would like to run it as part of our EC codebase presubmit), so it'd be ideal if it would just work.

Reply:
> The current code is considering those as well

I restored this functionality; thanks.

> I'm also interested in having this check usable in an external repository

Fair enough, but that's out of scope for what I'm trying to do since the current code isn't doing that.